### PR TITLE
Refactor custom hooks

### PR DESCRIPTION
**Pull request by mvastarelli**
Introduced a number of hooks to cut down on boilerplate.  Added tests to bump up code coverage throughout the app.  Also eliminated the need for global state via `MinerContext`.

### CODE REVIEW

File: src/renderer/hooks/observable.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  useEffect(() => {
+    const subscription = observable.subscribe(setCurrent);
+
+    return () => {
+      subscription.unsubscribe();
+    };
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
Comment (evan-cohen) [inline src/renderer/hooks/observable.ts:28]: Could this useEffect be replaced with a call to useObservable? 

```
useObservable (observable, setCurrent)
```

File: src/renderer/hooks/observable.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+export function useObservable<T>(observable: Observable<T>, handler: (value: T) => void) {
+  useEffect(() => {
+    const subscription = observable.subscribe(handler);
+
+    return () => {
+      subscription.unsubscribe();
+    };
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (evan-cohen) [inline src/renderer/hooks/observable.ts:12]: Are there unintended consequences of including `observable` and `handler` in the useEffect dependency array?

Ignoring that ESLint rule is often a code smell and in this case I'd think we'd _want_ to re-run the effect if the reference to the observable or handler function changed.

File: src/renderer/hooks/profile.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  useEffect(() => {
+    const configSubscription = watchers$.settings.subscribe((appSettings) => {
+      setProfile(appSettings.settings.defaultMiner);
+    });
+
+    return () => {
+      configSubscription.unsubscribe();
+    };
+  }, []);
```
Comment (evan-cohen) [inline src/renderer/hooks/profile.ts:21]: Could this also be a call to useObservable?

**Comment by mvastarelli** [comment]
Refactored hooks to use reuse them for other hooks.  Added custom hooks to `App.tsx`.

**Review by evan-cohen — APPROVED** [review]

**Pull request merged**